sys/cbor/cbor.c

```diff
@@ -92,6 +96,18 @@
 #endif
 
 #ifndef CBOR_NO_FLOAT
+
+/* pack to force aligned access on ARMv7 (buggy GCC) */
+#pragma GCC diagnostic error "-Wcast-align"
```

Comment:
I did not look properly into the change, but I am 90% sure that the warning here exposes a but, and should not be ignored.